### PR TITLE
drivers: gpio_gecko: convert to DT_INST defines

### PR DESCRIPTION
**Pull request by mnkp**
Convert the driver to use DT_INST_ defines, update all dependent dts, soc and board files.

I've tested the changes on efr32_slwstk6061a (efr32fg1p), efr32_radio_brd4250b (efr32fg1p) and efr32_radio_brd4104a (efr32bg13p) boards using `tests/drivers/gpio/gpio_api_1pin/` as well as `samples/subsys/logging/logger/` to verify UART and SWO output.

It would be great if someone could test also other boards.

### CODE REVIEW

File: boards/arm/efm32gg_stk3701a/board.h [hunk 1]
```diff
@@ -9,18 +9,18 @@
 #define __INC_BOARD_H
 
 /* This pin is used to enable the serial port using the board controller */
-#define BC_ENABLE_GPIO_NAME		DT_GPIO_GECKO_PORTE_NAME
+#define BC_ENABLE_GPIO_NAME		"GPIO_E"
```
Comment (mnkp) [inline boards/arm/efm32gg_stk3701a/board.h]: This should come from DTS. However, I would prefer to fix it in subsequent PR as it requires adding a new binding.

File: dts/arm/silabs/efm32_jg_pg_12b.dtsi [hunk 2]
```diff
@@ -132,75 +132,93 @@
...
 				label = "GPIO_I";
 				gpio-controller;
 				#gpio-cells = <2>;
+				status = "disabled";
 			};
 
-			gpioj: gpio@4000a1c0 {
-				compatible = "silabs,efm32-gpio-port";
-				reg = <0x4000a1c0 0x30>;
+			gpioj: gpio@4000a1b0 {
```
Comment (mnkp) [inline dts/arm/silabs/efm32_jg_pg_12b.dtsi]: Register address modification is a bugfix. I should have submitted it in a separate commit, will do if requested.

**Review by chrta — APPROVED** [review]
Tested successfully on custom EFM32JG12B board, since i currently do not have access to other boards (should be available in a week).

**Review by galak — APPROVED** [review]

**Review by ioannisg — APPROVED** [review]

**Review by pabigot — APPROVED** [review]
efr32mg_sltb004a works fine on this branch.
